core/src/main/resources/org/apache/spark/ui/static/taskspages.js

```diff
@@ -0,0 +1,474 @@
+
+    "file-size-asc": function ( a, b ) {
+        a = ConvertDurationString( a );
+        b = ConvertDurationString( b );
+        return ((a < b) ? -1 : ((a > b) ? 1 : 0));
+    },
+
+    "file-size-desc": function ( a, b ) {
+        a = ConvertDurationString( a );
+        b = ConvertDurationString( b );
+        return ((a < b) ? 1 : ((a > b) ? -1 : 0));
+    }
+} );
+
+function createTemplateURI(appId) {
+    var words = document.baseURI.split('/');
+    var ind = words.indexOf("proxy");
+    if (ind > 0) {
+        var baseURI = words.slice(0, ind + 1).join('/') + '/' + appId + '/static/stagespage-template.html';
+        return baseURI;
+    }
+    ind = words.indexOf("history");
+    if(ind > 0) {
+        var baseURI = words.slice(0, ind).join('/') + '/static/stagespage-template.html';
+        return baseURI;
+    }
+    return location.origin + "/static/stagespage-template.html";
+}
+
+// This function will only parse the URL under certain formate
+// e.g. https://axonitered-jt1.red.ygrid.yahoo.com:50509/history/application_1502220952225_59143/stages/stage/?id=0&attempt=0
+function StageEndPoint(appId) {
+    var words = document.baseURI.split('/');
+    var words2 = document.baseURI.split('?');
+    var ind = words.indexOf("proxy");
+    if (ind > 0) {
+        var appId = words[ind + 1];
+        var stageIdLen = words2[1].indexOf('&');
+        var stageId = words2[1].substr(3, stageIdLen - 3);
+        var newBaseURI = words.slice(0, ind + 2).join('/');
+        return newBaseURI + "/api/v1/applications/" + appId + "/stages/" + stageId;
+    }
+    ind = words.indexOf("history");
+    if (ind > 0) {
+        var appId = words[ind + 1];
+        var attemptId = words[ind + 2];
+        var stageIdLen = words2[1].indexOf('&');
+        var stageId = words2[1].substr(3, stageIdLen - 3);
+        var newBaseURI = words.slice(0, ind).join('/');
+        if (isNaN(attemptId) || attemptId == "0") {
+            return newBaseURI + "/api/v1/applications/" + appId + "/stages/" + stageId;
+        } else {
+            return newBaseURI + "/api/v1/applications/" + appId + "/" + attemptId + "/stages/" + stageId;
+        }
+    }
+    var stageIdLen = words2[1].indexOf('&');
+    var stageId = words2[1].substr(3, stageIdLen - 3);
+    return location.origin + "/api/v1/applications/" + appId + "/stages/" + stageId;
+}
+
+function sortNumber(a,b) {
+    return a - b;
+}
+
+function quantile(array, percentile) {
+    index = percentile/100. * (array.length-1);
+    if (Math.floor(index) == index) {
+    	result = array[index];
+    } else {
+        var i = Math.floor(index);
+        fraction = index - i;
+        result = array[i];
+    }
+    return result;
+}
+
+$(document).ready(function () {
+    $.extend($.fn.dataTable.defaults, {
+        stateSave: true,
+        lengthMenu: [[20, 40, 60, 100, -1], [20, 40, 60, 100, "All"]],
+        pageLength: 20
+    });
+
+    $("#showAdditionalMetrics").append(
```

Comment:
If this is just adding static html we can probably leave it in `StagePage.scala`

Reply:
The reason I wrote that piece of code in javascript is because I thought if in future, the data-column indices needed to be rearranged dynamically, it would make more sense to do it in javascript. But, if you feel otherwise, I shall move it to html.